registry/registry.go

```diff
@@ -55,14 +55,14 @@ func NewInsecure(registryUrl, username, password string) (*Registry, error) {
 		},
 	}
 
-	return newFromTransport(registryUrl, username, password, transport, Log)
+	return newFromTransport(registryUrl, username, password, transport, false, Log)
 }
 
 /*
  * Create a new Registry, as with New, that uses the provided http.RoundTripper as transport.
  */
-func NewWithTransport(registryUrl, username, password string, transport http.RoundTripper) (*Registry, error) {
-	return newFromTransport(registryUrl, username, password, transport, Log)
+func NewWithTransport(registryUrl, username, password string, transport http.RoundTripper, disableBasicAuth bool) (*Registry, error) {
```

Comment:
I'd prefer not to change the signature for an exported function, perhaps create a new function that has the behaviour you're looking for?